src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ApplicationPropertyPage/CompoundTargetFrameworkValueProvider.cs

```diff
@@ -208,9 +212,17 @@ private async Task<string> ComputeValueAsync(ComplexTargetFramework complexTarge
             return targetFrameworkAlias;
         }
 
-        private static bool IsNetCore5OrHigher(string targetFrameworkAlias)
+        private static bool IsNetCore5OrHigher(string targetFrameworkAlias, string? targetFrameworkIdentifier)
         {
-            return targetFrameworkAlias.Contains("5.0") || targetFrameworkAlias.Contains("6.0") || targetFrameworkAlias.Contains("7.0");
+            // Converts the target framework alias to a double and checks if it's greater than or equal to 5.0.
+            // First, remove any non-numeric characters. Example: net5.0 => 5.0
+            targetFrameworkAlias = Regex.Replace(targetFrameworkAlias, "[^0-9.]", string.Empty);
```

Comment:
It's not safe to use the alias here. People can make up their own aliases to mean whatever they want.

The correct way to check the framework version is to compare `TargetFrameworkIdentifier` and `TargetFrameworkVersion`. You're already checking the former, and should add the latter (which I believe will start with a `v`, such as `v5.0`, so you'll have to trim that off).

Reply:
We discussed the approach offline some time ago. At the end, we realized the limitations of the current approach blocks us from using the `ConfigurationGeneral` rules mentioned here, since at this point, they contain data for the previous target framework value, not the one being set. 

For now, I'll focus on improving our approach of detecting when a project is .NET core 5 or higher with the `TargetFrameworkAlias`. #7958 captured this future improvements.